extensions/wikia/WallNotifications/WallNotificationsEveryone.class.php

```diff
@@ -1,31 +1,57 @@
 <?php
 
-// TODO: Comments!
+/**
+ * Class WallNotificationsEveryone that implements wall notifications queue.
+ * It's called by WallMessage::setNotifyEveryone method - triggered by notifyEveryoneSave AJAX request.
+ *
+ * ## How to call it
+ *
+ * $notif = WallNotificationEntity::createFromRev( $rev );
+ * $wne = new WallNotificationsEveryone();
+ * $wne->addNotificationToQueue( $notif );
+ *
+ * ## Database queries
+ *
+ * This class uses wall_notification_queue and wall_notification_queue_processed tables on dataware database.
+ *
+ * entity_id column is composed of "<revision ID>_<wiki ID>"
+ *
+ * ## DB schema
+ *
+ *  - wall_notification_queue table stores the date of the last edit (event_date column) of Message Walls (page_id + entity_key columns) across all wikis
+ *  - wall_notification_queue_processed stores per-user notifications that were added to wall_notifications
+ *
+ * When WallNotificationsExternalController::getUpdateCounts is handled processQueue() method moves entries from wall_notification_queue to wall_notification_queue_processed table (using processEntities method).
+ * All entity_key values for a given wiki (taken from wall_notification_queue) are insrted into wall_notification_queue_processed table (with a specific user_id set) and wall_notification entry is created.
+ * Hence wall_notification_queue_processed table is used to mark entities which user was already notified about.
+ *
+  */
 class WallNotificationsEveryone extends WallNotifications {
 
 	const DELETE_IDS_BATCH_SIZE = 100;
 
+	const WALL_NOTIFICATIONS_QUEUE_TABLE = 'wall_notification_queue';
+	const WALL_NOTIFICATIONS_QUEUE_PROCESSED_TABLE = 'wall_notification_queue_processed';
+
 	public function __construct() {
 		parent::__construct();
 
-		global $wgCityId, $wgWikiaEnvironment;
-
-		$this->app = F::app();
+		global $wgCityId;
 		$this->cityId = $wgCityId;
 	}
 
 	/**
-	 * Adds notification to the notification queue
+	 * Adds notification to the notification queue. Called by WallMessage::setNotifyEveryone
 	 *
 	 * @param WallNotificationEntity $entity
 	 */
 	public function addNotificationToQueue( WallNotificationEntity $entity ) {
 		wfProfileIn( __METHOD__ );
 
 		$key = $entity->id;
-		$pageId = $entity->data->title_id;
+		$pageId = $entity->data->title_id; // ID of MessageWall title, each Wall has a unique entry in this table
 
-		$this->getDB( true )->replace( 'wall_notification_queue', '',
+		$this->getDB( true )->replace( self::WALL_NOTIFICATIONS_QUEUE_TABLE, '',
 			[
 				'wiki_id' => $this->cityId,
 				'entity_key' => $key,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "entity_key" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)